core/trino-main/src/main/java/io/trino/sql/planner/TranslationMap.java

```diff
@@ -646,6 +650,42 @@ public Expression rewriteTryExpression(TryExpression node, Void context, Express
                 return coerceIfNecessary(node, call);
             }
 
+            @Override
+            public Expression rewriteLikePredicate(LikePredicate node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+            {
+                Optional<SymbolReference> mapped = tryGetMapping(node);
+                if (mapped.isPresent()) {
+                    return coerceIfNecessary(node, mapped.get());
+                }
+
+                Expression value = treeRewriter.rewrite(node.getValue(), context);
+                Expression pattern = treeRewriter.rewrite(node.getPattern(), context);
+                Optional<Expression> escape = node.getEscape().map(e -> treeRewriter.rewrite(e, context));
+
+                FunctionCall patternCall;
+                if (escape.isPresent()) {
+                    patternCall = FunctionCallBuilder.resolve(session, plannerContext.getMetadata())
+                            .setName(QualifiedName.of(LIKE_PATTERN_FUNCTION_NAME))
+                            .addArgument(analysis.getType(node.getPattern()), pattern)
+                            .addArgument(analysis.getType(node.getEscape().get()), escape.get())
+                            .build();
+                }
+                else {
+                    patternCall = FunctionCallBuilder.resolve(session, plannerContext.getMetadata())
+                            .setName(QualifiedName.of(LIKE_PATTERN_FUNCTION_NAME))
+                            .addArgument(analysis.getType(node.getPattern()), pattern)
+                            .build();
+                }
+
+                FunctionCall call = FunctionCallBuilder.resolve(session, plannerContext.getMetadata())
```

Comment:
@martint this change causes the following impact:
Since now we transform `LikePredicate` into a `FunctionCall`, `io.trino.sql.tree.ExpressionRewriter#rewriteLikePredicate` is not called anymore and therefore, for example, doesn't turn `char(1) LIKE 'a'` into a `ComparisonExpression`. `DomainTranslator` doesn't process `like` of char type so now we push down such filter as a `ConnectorExpression` instead of `TupleDomain`